File: src/main/java/com/design/platform/resourceplatform/transfer/results/PageHolder.java
package com.design.platform.resourceplatform.transfer.results;

import org.springframework.data.domain.Page;

import java.util.List;

public class PageHolder<T> {

    public List<T> content;

    public int pageCount;
    public long itemCount;

    public int page;
    public int size;

    public PageHolder(Page<T> page) {
        content = page.getContent();
        pageCount = page.getTotalPages();
        itemCount = page.getTotalElements();
        this.page = page.getNumber();
        size = page.getSize();
    }

    public static <T> PageHolder<T> FromPage(Page<T> page) {
        return new PageHolder<>(page);
    }
}

File: src/main/java/com/design/platform/resourceplatform/utils/PageHolder.kt
package com.design.platform.resourceplatform.utils

import org.springframework.data.domain.Page

class PageHolder<T>(page: Page<T>) {
    var content: List<T> = page.content
    var pageCount: Int = page.totalPages
    var itemCount: Long = page.totalElements
    var page: Int = page.number
    var size: Int = page.size
}

val <T> Page<T>.auto: PageHolder<T>
    get() {
        return PageHolder(this)
    }
